lib/http/server.go

```diff
@@ -293,12 +293,19 @@ func NewServer(ctx context.Context, options ...Option) (*Server, error) {
 }
 
 func (s *Server) initAuth() {
+	s.usingAuth = false
+
 	if s.auth.CustomAuthFn != nil {
 		s.usingAuth = true
 		s.mux.Use(MiddlewareAuthCustom(s.auth.CustomAuthFn, s.auth.Realm))
 		return
 	}
 
+	if s.auth.MTLS {
```

Comment:
This depends on client CA being provided too

Reply:
Enabling this without enabling client CA will result in unauthorized requests, but should we refuse to enable the option here in that case? And if so, do we warn about it? Perhaps we should error and exit earlier if cert usernames are enabled but client-ca is not?

If we decide that certs as usernames should be the default when client-ca is provided then this logic will need to change in any case.

Reply:
If we decide to have a flag then we should error if `--client-ca` is not provided.

Reply:
Resolved by removing the flag and enabling by default.